numpy/linalg/tests/test_linalg.py

```diff
@@ -333,94 +315,107 @@ def _make_strided_cases():
                 new_cases.append(new_case)
     return new_cases
 
+
 CASES += _make_strided_cases()
 
 
 #
 # Test different routines against the above cases
 #
+class LinalgTestCase(object):
+    TEST_CASES = CASES
 
-def _check_cases(func, require=set(), exclude=set()):
-    """
-    Run func on each of the cases with all of the tags in require, and none
-    of the tags in exclude
-    """
-    for case in CASES:
-        # filter by require and exclude
-        if case.tags & require != require:
-            continue
-        if case.tags & exclude:
-            continue
+    def check_cases(self, func, require=set(), exclude=set()):
```

Comment:
Is there any point passing in `func`, now that it's always `self.do`?

Reply:
None at all. A bit beyond the PR, but I'll fix it anyway.